contracts/src/v0.8/functions/tests/1_0_0/FunctionsSubscriptions.t.sol

```diff
@@ -123,66 +440,578 @@ contract FunctionsSubscriptions_createSubscription is FunctionsOwnerAcceptTermsO
+    assertEq(flags, flagsToSet);
+  }
 }
 
 /// @notice #getFlags
-contract FunctionsSubscriptions_GetFlags {
+contract FunctionsSubscriptions_GetFlags is FunctionsSubscriptionSetup {
+  function test_GetFlags_Success() public {
+    // Set flags
+    bytes32 flagsToSet = bytes32("1");
+    s_functionsRouter.setFlags(s_subscriptionId, flagsToSet);
 
+    // Send as stranger
+    vm.stopPrank();
+    vm.startPrank(STRANGER_ADDRESS);
+
+    bytes32 flags = s_functionsRouter.getFlags(s_subscriptionId);
+    assertEq(flags, flagsToSet);
+  }
 }
 
 /// @notice #timeoutRequests
-contract FunctionsSubscriptions_TimeoutRequests {
+contract FunctionsSubscriptions_TimeoutRequests is FunctionsClientRequestSetup {
+  function test_TimeoutRequests_RevertIfPaused() public {
+    s_functionsRouter.pause();
 
+    vm.expectRevert("Pausable: paused");
+    FunctionsResponse.Commitment[] memory commitments = new FunctionsResponse.Commitment[](1);
+    commitments[0] = s_requestCommitment;
+    s_functionsRouter.timeoutRequests(commitments);
+  }
+
+  function test_TimeoutRequests_RevertInvalidRequest() public {
+    // Modify the commitment so that it doesn't match
+    s_requestCommitment.donFee = 123456789;
+    FunctionsResponse.Commitment[] memory commitments = new FunctionsResponse.Commitment[](1);
+    commitments[0] = s_requestCommitment;
+    vm.expectRevert(FunctionsSubscriptions.InvalidCalldata.selector);
+    s_functionsRouter.timeoutRequests(commitments);
+  }
+
+  function test_TimeoutRequests_RevertIfTimeoutNotExceeded() public {
+    vm.expectRevert(FunctionsSubscriptions.TimeoutNotExceeded.selector);
+    FunctionsResponse.Commitment[] memory commitments = new FunctionsResponse.Commitment[](1);
+    commitments[0] = s_requestCommitment;
+    s_functionsRouter.timeoutRequests(commitments);
+  }
+
+  event RequestTimedOut(bytes32 indexed requestId);
+
+  function test_TimeoutRequests_Success() public {
+    uint64 consumerCompletedRequestsBefore = s_functionsRouter
+      .getConsumer(address(s_functionsClient), s_subscriptionId)
+      .completedRequests;
+
+    // topic0 (function signature, always checked), NOT topic1 (false), NOT topic2 (false), NOT topic3 (false), and data (true).
+    bool checkTopic1 = false;
+    bool checkTopic2 = false;
+    bool checkTopic3 = false;
+    bool checkData = true;
+    vm.expectEmit(checkTopic1, checkTopic2, checkTopic3, checkData);
+    emit RequestTimedOut(s_requestId);
+
+    // Jump ahead in time past timeout timestamp
+    vm.warp(s_requestCommitment.timeoutTimestamp + 1);
+
+    FunctionsResponse.Commitment[] memory commitments = new FunctionsResponse.Commitment[](1);
+    commitments[0] = s_requestCommitment;
+    s_functionsRouter.timeoutRequests(commitments);
+
+    // Releases blocked balance and increments completed requests
+    uint96 subscriptionBlockedBalanceAfter = s_functionsRouter.getSubscription(s_subscriptionId).blockedBalance;
+    assertEq(0, subscriptionBlockedBalanceAfter);
```

Comment:
We could also add a check to confirm the blocked balance was added back to the subscription balance.

Reply:
"blocked" balance is never directly modified on the subscription balance
By checking that `blockedBalance = 0`, that would be the check that you are talking about